kasa/discover.py

```diff
@@ -224,11 +228,25 @@ async def do_discover(self) -> None:
         _LOGGER.debug("[DISCOVERY] %s >> %s", self.target, Discover.DISCOVERY_QUERY)
         encrypted_req = XorEncryption.encrypt(req)
         sleep_between_packets = self.discovery_timeout / self.discovery_packets
+
+        key_payload = {"params": {"rsa_key": self.keypair.get_public_pem().decode()}}
+
+        encoded_payload = json_dumps(key_payload).encode()
+        version = 2
+        nonce = 1337
+        initial_crc = 0x5A6B7C8D  # https://labs.withsecure.com/advisories/tp-link-ac1750-pwn2own-2019
```

Comment:
Probably a good idea to move the magics into classvar(?) consts? Also, we probably should consider creating the key-pair only once per library initialization?

Reply:
Done